### DISCUSSION TITLE
Getting MODE to work with self made data

### DISCUSSION THREAD

**Discussion by jcorner1** [dtcenter/METplus#2002] [File I/O] [answered]
I would like to note this is a bit of a time crunch, but I essentially have a file that is almost the exact same as (hard press to find the difference at this point from the working data) as the original data which is working fine in MODE. I have tried numerous different methods to try and get this data to work with MODE and feel that I am super close and might be missing something super basic or a minuet difference in the config, netcdf file, etc. As I stated before, this is a bit of a time crunch but if it can get working that would be great! I will send my data through the ftp server in the corner_data directory. Thanks!

**Comment by georgemccabe** [comment]
Hi @jcorner1, the large files you uploaded to the FTP are being corrupted again, likely due to the same issues we saw in #1924. If you are generating your own files that are meant to match the format of files you have previously read into MET, I would recommend running `ncdump -h {filename}` on both the file you created and the original to print the header information, pipe each result to a text file, then run the `diff` utility to compare the two text files. MET can read CF-compliant NetCDF files, so there may be some missing attributes that are required. If you are unable to track down the differences you could send us the text files that contain the header info. If you need to send the full files, you could try using Google Drive again, but please include the original file as well so I can compare it to your file.

We often recommend running plot_data_plane on a new file to see if it can be read by the other MET tools. You could run that with an increased verbosity to see if that gives you any other clues as to what is missing from the file. The command would look something like this:

`plot_data_plane mrms_lcref202006050000.nc4 out.ps 'name="mrms_lcref"; level="(*,*)";' -v 5`

You could also try increasing the verbosity of the MODE log output by setting `LOG_MODE_VERBOSITY` to an integer up to 10 (the default level is 2) in the METplus config file to see if you receive additional useful information in the logs.

Please note that many of our staff are still out of the office and others are preparing for a conference next week, so the support we can provided this week is limited, but we will do our best to help given the circumstances.

**Reply by jcorner1** [reply]
Hi @georgemccabe,

I have put all my files in this folder (https://drive.google.com/drive/folders/1diSpbv-0t3wNQEHwoBUqZV6FYsLngfZa?usp=share_link). I have tried matching the attributes the best as possible. I did get it to match but I don't think those data are uploaded in there. I'm just not sure what's giving it the issues.

**Reply by georgemccabe** [reply]
Hi @jcorner1, I only see 1 MRMS file in the folder, which I assume is the file you created yourself. Please also upload the original MRMS file that you are trying to mimic so I can compare them. Thanks.

**Reply by jcorner1** [reply]
Hi @georgemccabe,

I have placed the original mrms file in there. furthermore, I have changed the new one to be the more correct file although it basically changes very little

**Comment by georgemccabe** [comment]
Thanks for sending these files. The file ending with \_OG could be read into MET because it has a global attribute to note the CF conventions:

```
// global attributes:
                :Conventions = "CF-1.6" ;
```

It looks like the file you created has this info as an attribute for the mrms_lcref field instead:

`                mrms_lcref:Conventions = "CF-1.6" ;`

You can get around this by explicitly setting the file type with `file_type=NETCDF_NCCF;` to tell MET that the file should be interpreted as a CF-compliant NetCDF file.

`/usr/local/met/bin/plot_data_plane mrms_lcref202006050000.nc4 out.ps 'name="mrms_lcref"; level="(*,*)"; file_type=NETCDF_NCCF;' -v 5`

You can set this in your METplus config file like this:

`OBS_VAR1_OPTIONS = file_type=NETCDF_NCCF;`

The above Plot-Data-Plane command ran successfully, but the log output showed that the timing information was not read properly.

```
DEBUG 4: NcCfFile::open() -> could not extract valid time from the "time" variable.
DEBUG 4: NcCfFile::open() -> could not extract valid time from file name.
WARNING:
WARNING: NcCfFile::open() -> could not determine the valid time, using 0.
WARNING:
DEBUG 4: NcCfFile::open() -> could not extract init time from the "forecast_reference_time" variable.
DEBUG 4: NcCfFile::open() -> could not extract init time from file name.

...

DEBUG 4:    valid time: 19700101_000000
DEBUG 4:    lead time: 000000
DEBUG 4:    init time: 19700101_000000
DEBUG 4:    accum time: 000000
```

This means that MET will not be able to properly match the data with other files at the correct times. The \_OG file you sent also did not include any timing information.

I found another MRMS file you had sent for #1924 named mrms_lcref2022-11-04T21_00_00.nc4 that does have timing information:

```
        int64 time ;
                time:long_name = "initial time of forecast" ;
                time:standard_name = "forecast_reference_time" ;
                time:units = "seconds since 1970-01-01" ;
                time:calendar = "proleptic_gregorian" ;

...

 time = 1559001600 ;
```

Adding this variable to your NetCDF file should fix this issue. It is possible to override the metadata from the input in MET (see [this page](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html?highlight=set_attr#fcst) of the documentation and search for "set_attr") to set the init or valid time, but I would recommend setting it properly in the file you are creating.

**Reply by jcorner1** [reply]
Hi @georgemccabe,

Sorry for the long wait for a reply... AMS takes up a lot of energy. Since then, I have gotten pretty close to fixing the issues I had before. I will put my data in the same google drive folder as I did previously (https://drive.google.com/drive/folders/1diSpbv-0t3wNQEHwoBUqZV6FYsLngfZa?usp=share_link). the original files have OG in them. Unable to find a difference in the files at this point except obviously the changed reflectivity data and HRRR raw output is a grib2 and the new is nc4. Let me know if there is any information MET is missing. I have attached the log and config file to help.

[metplus.log.20230119220523.txt](https://github.com/dtcenter/METplus/files/10463095/metplus.log.20230119220523.txt)
[MODE_HRRR_FCST_RADAR_OBS.txt](https://github.com/dtcenter/METplus/files/10463648/MODE_HRRR_FCST_RADAR_OBS.txt)

**Reply by georgemccabe** [reply]
Hi @jcorner1,

The MRMS data can be read into MET without issue except for the lack of time information that I mentioned in my response earlier.

It looks like it is interpreting the HRRR data as lat/lon instead of lambert conformal because the dimensions of the variables are (y,x) and not (lat, lon). The original grib2 data can be read into MET directly and the grid information will be read properly.

**Reply by jcorner1** [reply]
Hi @georgemccabe,

Focusing on the data I created for HRRR, I've tried a few different ways of creating the .nc4 file and appeared to get a different error in the log file which I would conclude to be a step in the right direction. The new error is

`ERROR  : StringArray::operator[](int) const -> range check error!`

Let me know if there is a way to remedy this issue. Furthermore, my main issue with the HRRR data is I'm using copy to create a replica of the original file for the purpose of changing the reflectivity data and then running it through the MODE wrapper. My confusion is why this method does not work as I have worked every detail to be the same as the raw data but there is a different result. I've looked for other avenues or even scripts to create working data. If you potentially know of another route to go, please let me know. Thanks!

*EDIT*
I have uploaded the new data in the google drive (https://drive.google.com/drive/folders/1diSpbv-0t3wNQEHwoBUqZV6FYsLngfZa?usp=sharing)

**Comment by georgemccabe** [comment, marked as answer]
Hi @jcorner1, running your new file through plot_data_plane with verbosity set to 6 shows a number of log messages that identify missing information.

`/usr/local/met/bin/plot_data_plane hrrr_t20200602T0600z_wrfsfcf18.nc4 out.ps 'name="refc"; level="(*,*)";' -v 6`

> DEBUG 4: NcCfFile::open() -> could not extract init time from the "forecast_reference_time" variable.
> DEBUG 4: NcCfFile::open() -> could not extract init time from file name.
> DEBUG 4: NcCfFile::get_grid_from_dimensions() -> The coordinate variable "lat" does not exist.
> DEBUG 4: NcCfFile::get_grid_from_dimensions() -> The coordinate variable "lon" does not exist.
> DEBUG 4: NcCfFile::get_grid_from_dimensions() -> The coordinate variable "x" does not exist.
> DEBUG 4: NcCfFile::get_grid_from_dimensions() -> The coordinate variable "y" does not exist.
> DEBUG 6:
> DEBUG 6: NcCfFile::get_grid_from_coordinates() -> collect GRID info from "hail".
> DEBUG 6:
> DEBUG 4:
> DEBUG 4: NcCfFile::get_grid_from_coordinates() -> unknown units [1] for the coordinate variable [atmosphere]
> DEBUG 4:
> DEBUG 4:
> DEBUG 4: NcCfFile::get_grid_from_coordinates() -> unknown units [hours] for the coordinate variable [step]
> DEBUG 4:
> ERROR  :
> ERROR  : StringArray::operator[](int) const -> range check error!
> ERROR  :

Instead of using an external tool to copy and modify the data, I would recommend writing a Python Embedding script to read the grib data and make any modifications needed in the script. You can use this script to pass the data into the MET tools. Here are some resources:

[Python Embedding for 2D Data](https://met.readthedocs.io/en/latest/Users_Guide/appendixF.html#python-embedding-for-2d-data) from the MET User's Guide
[MODE Python Embedding use case example](https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/MODE/MODE_python_embedding.html#sphx-glr-generated-met-tool-wrapper-mode-mode-python-embedding-py)
[Example Python Embedding script that uses pygrib Python package](https://github.com/dtcenter/METplus/blob/main_v5.0/parm/use_cases/model_applications/medium_range/TCStat_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead_PyEmbed_Multiple_Diagnostics/gfs_ivt_fcst.py)
[Example Python Embedding script that uses cfgrib Python package](https://github.com/dtcenter/METplus/blob/main_v5.0/parm/use_cases/model_applications/medium_range/TCStat_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead_PyEmbed_Multiple_Diagnostics/gfs_sept_fcst.py)

EDIT: If necessary you can also use Python Embedding to pass in the data from the NetCDF file you created and explicitly set the attributes needed to describe the data to MET. However, I think it would be better to read the GRIB2 data directly into the Python Embedding script, perform any modifications you need, and pass the data to MET.

Also, [this page](https://dtcenter.org/community-code/model-evaluation-tools-met/sample-analysis-scripts) contains many Python Embedding examples with sample data, scripts, and calls to plot_data_plane. We always recommend using plot_data_plane to generate an image to verify that MET is reading the data in the correct location on the grid.

**Reply by jcorner1** [reply]
Hi @georgemccabe,

I'm having an issue trying to get the plot_data_plane command to work. I keep getting the error `/bin/bash: plot_data_plane: command not found` from the command line. I found the wrapper.py file for it, using it with python did not do turned errors, but that's not what you did, so that was kind of expected. I did navigate to the correct folder and run the plot_data_plane command, but that did not change anything. What am I missing here? Do I have to build the command despite having MET installed?

I used the scripts you provided and made the data into a text file named met_data as well passed the attributes as attrs so I feel like we're on the right track there. I'll send my notebook and txt file along in the google drive (https://drive.google.com/drive/folders/1diSpbv-0t3wNQEHwoBUqZV6FYsLngfZa?usp=sharing) to see if you can spot any issues ahead of the getting the command to work. Thanks!

**Reply by georgemccabe** [reply]
plot_data_plane should be available in the bin directory where MET is installed. You may need to provide the full path to the executable if it is not in your path. If you aren't sure where it is the directory that contains the bin directory should be set to MET_INSTALL_DIR in your METplus config.

**Reply by jcorner1** [reply]
Thanks, @georgemccabe for the help. I didn't check MET but only METplus.

I have a bit of confusion as I don't understand if the data only has to be declared in python before running the command which is what the documentation seems to suggest from what I'm reading, however I don't see an example of this in the documentation. Is this possible and is there an example to help with that? If not, I have loaded it into a txt file and have the attributes saved in a dictionary. Unsure of where the attribute information is to go as it is not in the documentation from what I see. The code and txt file are in the google drive folder (https://drive.google.com/drive/folders/1diSpbv-0t3wNQEHwoBUqZV6FYsLngfZa?usp=sharing). Thanks for the help!

**Reply by georgemccabe** [reply]
You will need to write a python script (.py file) that contains the logic to read the data and set the met_data and attrs variables. Then you will pass that script to the MET command. The [Sample Analysis Scripts](https://dtcenter.org/community-code/model-evaluation-tools-met/sample-analysis-scripts) page on the MET webpage contains many examples of calls to plot_data_plane that include the command to run, the Python Embedding script that is called, and the input data that is passed to the script.

**Reply by jcorner1** [reply]
Hi @georgemccabe,

That explanation helped a lot. I was misunderstanding how the script was used by MET within the command. Now I better understand and I have used some of the examples codes to do the same thing with my example. I did also try one of the examples from the page you provided (It was the NRL Binary surface data one I believe. Anyways, when I run my example using the `/gpfs/fs1/home/ac.jcorner/NIU/Masters_Thesis/MET_10.1.2/bin/plot_data_plane PYTHON_NUMPY test_out.ps 'name="Read_HRRR_Storm_Mode.py /lcrc/project/rainfall/jcorner/MODE_data/202006300000/isolated_cell/hrrr_t20200629T1500z_wrfsfcf09.nc4";',` command, I have an error/warning message outputted. It looks like this:

```
DEBUG 1: Opening data file: PYTHON_NUMPY
/home/ac.jcorner/anaconda3/envs/data/lib/python3.9/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.8.18 when it was built against 1.12.2, this may cause problems
  _warn(("h5py is running against HDF5 {0} when it was built against {1}, "
Warning! ***HDF5 library version mismatched error***
The HDF5 header files used to compile this application do not match
the version used by the HDF5 library to which this application is linked.
Data corruption or segmentation faults may occur if the application continues.
This can happen when an application was compiled by one version of HDF5 but
linked with a different version of static or shared HDF5 library.
You should recompile the application or check your shared library related
settings such as 'LD_LIBRARY_PATH'.
You can, at your own risk, disable this warning by setting the environment
variable 'HDF5_DISABLE_VERSION_CHECK' to a value of '1'.
Setting it to 2 or higher will suppress the warning messages totally.
Headers are 1.12.2, library is 1.8.18
	    SUMMARY OF THE HDF5 CONFIGURATION
	    =================================

General Information:
-------------------
		   HDF5 Version: 1.8.18
		  Configured on: Wed Nov  2 18:06:20 CDT 2022
		  Configured by: ac.jcorner@beboplogin2
		 Configure mode: production
		    Host system: x86_64-unknown-linux-gnu
	      Uname information: Linux beboplogin2 3.10.0-1160.62.1.el7.x86_64 #1 SMP Tue Apr 5 16:57:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
		       Byte sex: little-endian
		      Libraries: static, shared
	     Installation point: /gpfs/fs1/home/ac.jcorner/NIU/Masters_Thesis/MET_10.1.2/external_libs

Compiling Options:
------------------
               Compilation Mode: production
                     C Compiler: /usr/bin/gcc ( gcc (GCC) 4.8.5 20150623 )
                         CFLAGS: -fPIC
                      H5_CFLAGS: -std=c99 -pedantic -Wall -Wextra -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wnested-externs -Winline -Wfloat-equal -Wmissing-format-attribute -Wmissing-noreturn -Wpacked -Wdisabled-optimization -Wformat=2 -Wunreachable-code -Wendif-labels -Wdeclaration-after-statement -Wold-style-definition -Winvalid-pch -Wvariadic-macros -Winit-self -Wmissing-include-dirs -Wswitch-default -Wswitch-enum -Wunused-macros -Wunsafe-loop-optimizations -Wc++-compat -Wstrict-overflow -Wlogical-op -Wlarger-than=2048 -Wvla -Wsync-nand -Wframe-larger-than=16384 -Wpacked-bitfield-compat -Wstrict-overflow=5 -Wjump-misses-init -Wdouble-promotion -Wsuggest-attribute=const -Wtrampolines -Wstack-usage=8192 -Wvector-operation-performance -Wsuggest-attribute=pure -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -O3
                      AM_CFLAGS: 
                       CPPFLAGS: -I/gpfs/fs1/home/ac.jcorner/NIU/Masters_Thesis/MET_10.1.2/external_libs/include
                    H5_CPPFLAGS: -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112L   -DNDEBUG -UH5_DEBUG_API
                    AM_CPPFLAGS: 
               Shared C Library: yes
               Static C Library: yes
  Statically Linked Executables: no
                        LDFLAGS: -L/gpfs/fs1/home/ac.jcorner/NIU/Masters_Thesis/MET_10.1.2/external_libs/lib:
                     H5_LDFLAGS: 
                     AM_LDFLAGS: 
 	 	Extra libraries: -lz -ldl -lm 
 		       Archiver: ar
 		 	 Ranlib: ranlib
 	      Debugged Packages: 
		    API Tracing: no

Languages:
----------
                        Fortran: no

                            C++: no

Features:
---------
                  Parallel HDF5: no
             High Level library: yes
                   Threadsafety: no
            Default API Mapping: v18
 With Deprecated Public Symbols: yes
         I/O filters (external): deflate(zlib)
                            MPE: no
                     Direct VFD: no
                        dmalloc: no
Clear file buffers before write: yes
           Using memory checker: no
         Function Stack Tracing: no
      Strict File Format Checks: no
   Optimization Instrumentation: no
Bye...
```
I noticed in the beginning of the message that it states that a library is built and it is different than what it is being ran with. I'm not sure how to proceed if that is in fact the issue. For more information, I will put as much information as possible in the google drive. Thanks for all your help again!

**Reply by georgemccabe** [reply]
This issue is covered to discussion https://github.com/dtcenter/METplus/discussions/1979. You will need to set the MET_PYTHON_EXE environment variable to the full path of the version of python3 you want to use to run the Python Embedding command.

**Reply by jcorner1** [reply]
Hi @georgemccabe,

I tried all three of the methods listed, however none of them seemed to fix the issue. I set all of them in the command line as well as setting the `MET_PYTHON_EXE=/home/ac.jcorner/anaconda3/bin/python3` in the defualts.confg file in metplus. I did notice that the given example is ran with metplus stuff while this is just MET. Thanks!

**Reply by jcorner1** [reply]
Hi @georgemccabe,

Just a little update. I've tried a few different ways of adding the MET_PYTHON_EXE with no success, I also tried suppressing the HDF5 warning in the command line and still found it not to work and also still output the warning I posted earlier. here is the command I used `/gpfs/fs1/home/ac.jcorner/NIU/Masters_Thesis/MET_10.1.2/bin/plot_data_plane PYTHON_NUMPY test_out.ps 'name="Read_HRRR_Storm_Mode.py /lcrc/project/rainfall/jcorner/MODE_data/202006300000/isolated_cell/hrrr_t20200629T1500z_wrfsfcf09.nc4 user_env_vars.MET_PYTHON_EXE=/home/ac.jcorner/anaconda3/bin/python3";'` . I have tried setting the path outside of the field string, however it gives me error indicating I done the command wrong. Please let me know if there is a better way of doing this

**Reply by georgemccabe** [reply]
You will want to set MET_PYTHON_EXE as an environment variable in the shell before you call plot_data_plane. The user_env_vars functionality only works if you are calling run_metplus.py. Try this:

```
export MET_PYTHON_EXE=/home/ac.jcorner/anaconda3/bin/python3
/gpfs/fs1/home/ac.jcorner/NIU/Masters_Thesis/MET_10.1.2/bin/plot_data_plane PYTHON_NUMPY test_out.ps 'name="Read_HRRR_Storm_Mode.py /lcrc/project/rainfall/jcorner/MODE_data/202006300000/isolated_cell/hrrr_t20200629T1500z_wrfsfcf09.nc4
```

**Reply by jcorner1** [reply]
Hi @georgemccabe,

That worked in terms of getting further along in the process of bringing it into MET. Thanks! I'm getting a new error after running the shell script. It is posted below:

```
DEBUG 1: Opening data file: PYTHON_NUMPY
Python Script:  '/gpfs/fs1/home/ac.jcorner/NIU/Masters_Thesis/MET_10.1.2/share/met/wrappers/write_tmp_dataplane.py'
User Command:   'Read_HRRR_Storm_Mode /lcrc/project/rainfall/jcorner/MODE_data/202006300000/isolated_cell/hrrr_t20200629T1500z_wrfsfcf09.nc4'
Temporary File: '/tmp/tmp_met_nc_23267_0'
Trying file: Read_HRRR_Storm_Mode
Trying file: /lcrc/project/rainfall/jcorner/MODE_data/202006300000/isolated_cell/hrrr_t20200629T1500z_wrfsfcf09.nc4
Correct file found
Traceback (most recent call last):
  File "/gpfs/fs1/home/ac.jcorner/NIU/Masters_Thesis/MET_10.1.2/share/met/wrappers/read_tmp_dataplane.py", line 33, in <module>
    met_attrs['name'] = met_attrs['name_str']
KeyError: 'name_str'
WARNING:
WARNING: tmp_nc_dataplane() -> an error occurred importing module "read_tmp_dataplane"
WARNING:
ERROR  :
ERROR  : plot_data_plane -> trouble getting field "name="Read_HRRR_Storm_Mode.py /lcrc/project/rainfall/jcorner/MODE_data/202006300000/isolated_cell/hrrr_t20200629T1500z_wrfsfcf09.nc4";" from file "PYTHON_NUMPY"
ERROR  :

```
Let me know if it is a problem with my code or how the attributes are set or something else. Thanks!

**Reply by georgemccabe** [reply]
Do you have the 'name' attribute set in the attrs dictionary in your Read_HRRR_Storm_Mode.py script?

**Reply by jcorner1** [reply]
Hi @georgemccabe,

That fixed that error as well, thanks again! After adding the name, I have this error `Python3 Dict::lookup string(const char * key) -> value for key "type" not found`. I feel like we're getting close, so thank you with help on this project.

**Reply by georgemccabe** [reply]
The Python Embedding section of the MET User's Guide linked above outlines all of the attributes that you must set for Python Embedding. I would recommend reading through that section and set all of the variables that are described.

**Reply by jcorner1** [reply]
I got that done and got it plotted in MET, but I'm unsure of what some of these cf conventions mean. 
```
            'r_km' :
            'lon_orient' :
            'lat_pin' :
            'lon_pin' :
            'x_pin' :
            'y_pin' : 
```
I tried just inputting zeros for all of them, however that put the plot in a random location. I don't see anything within the metadata of the grib file that corresponds to those. Any help would be great!

**Reply by georgemccabe** [reply]
If you scroll down on the MET User's Guide page about 2D Python Embedding, it describes the grid info that needs to be set. There is a link to [Grid Specification Strings](https://met.readthedocs.io/en/latest/Users_Guide/appendixB.html#grid-specification-strings) that describes these values. These correspond to grid definitions of the various supported projections.

> D_km and R_km are the grid resolution and the radius of the Earth, both in kilometers.
> lon_orient is the orientation longitude of the grid.

lat/lon_pin is lat/lon_ll:
> lat_ll and lon_ll are the latitude and longitude, in degrees, of the lower left point of the grid.

x/y_pin are the x and y values of the lower left point of the grid. I believe these should be 0.0.

**Reply by jcorner1** [reply]
Hi @georgemccabe, I have pretty much gotten it working, however the HRRR plotted data is reflected upside down around the x-axis ( a storm is projected in Mexico despite it actually occurring in Minnesota). I have tried changing the lat_II and lon_II to the upper left point, however that still projects the object the wrong way in the same manner. Any Ideas? Thanks

**Reply by georgemccabe** [reply]
Hi @jcorner1, the ll values need to be the lower left corner. You will want to flip the data. This should work: `met_data = met_data[::-1]`

**Reply by jcorner1** [reply]
Hi @georgemccabe,

Thanks for the help so far. Pretty much almost done. I can run the data through MODE and get results. However inspection of the HRRR output from MET and the output from Matplotlib shows the actual values being about county south of MET is currently projecting it. Just to make sure I did compare the MRMS as well and there was no shift, so I think my input in attributes for HRRR would have to be incorrect at some point. I've looked it over a few times and can not seem to find anything that doesn't match. Is there a way to find the correct. Here are the attrs I have set which has a lot of the original left in:

```
attrs = {
        'valid': valid.strftime("%Y%m%d_%H%M%S"),
        'init':  init.strftime("%Y%m%d_%H%M%S"),
        'lead':  fhour,
        'accum': '00',
        'name': 'Reflectivity',
    
        'GRIB_name' : 'Maximum/Composite radar reflectivity',
        'GRIB_shortName' : 'refc',
        'GRIB_units' : 'dB',
        'long_name' : 'Maximum/Composite radar reflectivity',
        'units' : 'dB',
        'type' : 'Lambert Conformal',
        'level' : 'surface',
        
        'grid': {
            'GRIB_paramId' : 260390,
            'GRIB_dataType' : 'fc',
            'GRIB_numberOfPoints' : 1905141,
            'GRIB_typeOfLevel' : 'atmosphere',
            'GRIB_stepUnits' : 1,
            'GRIB_stepType' : 'instant',
            'GRIB_gridType' : 'lambert',
            'GRIB_DxInMetres' : 3000.0,
            'GRIB_DyInMetres' : 3000.0,
            'GRIB_LaDInDegrees' : 38.5,
            'GRIB_Latin1InDegrees' : 38.5,
            'GRIB_Latin2InDegrees' : 38.5,
            'GRIB_LoVInDegrees' : 262.5,
            'GRIB_NV' : 0,
            'GRIB_Nx' : 1799,
            'GRIB_Ny' : 1059,
            'GRIB_cfName' : 'unknown',
            'GRIB_cfVarName' : 'unknown',
            'GRIB_gridDefinitionDescription' : 'Lambert Conformal can be secant or tangent, conical or bipolar',
            'GRIB_iScansNegatively' : 0,
            'GRIB_jPointsAreConsecutive' : 0,
            'GRIB_jScansPositively' : 1,
            'GRIB_latitudeOfFirstGridPointInDegrees' : 21.138123,
            'GRIB_latitudeOfSouthernPoleInDegrees' : 0.0,
            'GRIB_longitudeOfFirstGridPointInDegrees' : 237.280472,
            'GRIB_longitudeOfSouthernPoleInDegrees' : 0.0,
            'GRIB_missingValue' : 9999,
            'name': 'Reflectivity',
            'type' : 'Lambert Conformal',
            'hemisphere' : 'N',
            'nx' : 1799,
            'ny' : 1059,
            'scale_lat_1' : 38.5,
            'scale_lat_2' : 38.5,
            'd_km' : 3.0, 
            #'r_km' : 6357.0,
            'r_km' : 6367.47,
            'lon_orient' : -95.0,
            'lat_pin' : 21.13812300000003,
            #'lat_pin' : 47.8386235,
            'lon_pin' : 237.28047200000003,
            #'lon_pin' : 225.90452027,
            'x_pin' : 0.0,
            'y_pin' : 0.0
            }
        }        
        

```

**Reply by georgemccabe** [reply]
I think you want to use the LoV value (262.5) as the lon_orient. Also, did the earth radius (r_km) value come from the file info? I noticed that was different than the example from the documentation (6371.2). The rest seem correct as far as I can tell. I would try adjusting those values and run through plot_data_plane to compare to the Matplotlib plots.

If you still can't get it to map properly, you could send an image of the matplotlib plot, the input file that was used, the Python Embedding script you are using, and the plot_data_plane command.

**Reply by jcorner1** [reply]
Hi @georgemccabe, that was it. I tried the LoV value at one point, but I put 265.0 instead of 262.5... they joys of coding and missing small yet simple details! Anyways, thanks so much for the help! I owe DTC a lot!!

**Reply by georgemccabe** [reply]
Great to hear! I am going to lock this discussion to prevent future posts. If you have any other questions or issues, please create a new discussion. Thanks!
